postgres/datadog_checks/postgres/statements.py

```diff
@@ -406,25 +455,58 @@ def _emit_pg_stat_statements_metrics(self):
         except psycopg2.Error as e:
             self._log.warning("Failed to query for pg_stat_statements count: %s", e)
 
+    # _apply_deltas expects normalized rows before any merging of duplicates.
+    # It takes the partial rows from pg_stat_statements and aggregates metric values
+    # for queryids that map to the same query_signature
+    def _apply_deltas(self, rows, metrics):
+        # Apply called queries to baseline_metrics
+        for row in rows:
+            query_signature = row['query_signature']
+            queryid = row['queryid']
+            if query_signature not in self._baseline_metrics:
+                self._baseline_metrics[query_signature] = {
+                    queryid: copy.copy(row)
+                }
+            else:
+                self._baseline_metrics[query_signature][queryid] = copy.copy(row)
```

Comment:
do we need to maintain a dict of `queryid` to `row` for every signature in `_baseline_metrics`? Can we aggregate the metrics here? `_baseline_metrics` can be a dict of `query_signature` with values be an `aggregated_row`.

Reply:
that should be possible, yea.

Reply:
I gave this a shot and this actually doesn't work. The main problematic scenario is when we have a set of `queryids` that map to the same `query_signature`, and only a subset of those are called between each check run. If we aggregate all of those query metrics together, we lose the contribution of the specific query id, so when the new values come in we don't know how to get to the proper new aggregated value.

Ex:

```
# Initial state - 40 calls from queryid 1, 60 calls from queryid 2:
{
  'abc': { calls: 100, ... }
}

# Next check run - queryid 1 called once, so it shows up as 41 calls:

{
  'abc': { calls: ??? }
}
```

`calls` should end up as 101 there, but to do that we need to know what queryid 2's call count is (and the same for all other metrics columns).